compiler/optimizer/Inliner.cpp

```diff
@@ -1797,7 +1797,7 @@ TR_InlinerBase::addGuardForVirtual(
    bool skipHCRGuardCreation = false;
 
    // addGuardForVirtual: create an HCRGuard after the original guard
-   if (!disableHCRGuards && comp()->getHCRMode() == TR::traditional && guard->_kind != TR_HCRGuard && !skipHCRGuardForCallee)
```

Comment:
Just for education and documentation purposes could you please explain what the different HCR modes are ? This would help someone reading the code understand what "traditional" means for example.